File: webapp/src/infrastructure/api/dto.ts
export type PollOptionResponse = {
  id: string;
  text: string;
  selected: boolean;
};

export type PollStatistics = {
  distribution: Record<string, number>;
};

export type PollResponse = {
  id: string;
  title: string;
  description: string | null;
  completed: boolean;
  can_edit: boolean;
  options: PollOptionResponse[];
  statistics: PollStatistics | null;
};

export type PollCompletion = {
  option_id: string;
};

export type ChangePasswordRequest = {
  old_password: string;
  new_password: string;
};

File: server/pollnow/dto.py
from typing import Optional, List, Dict
from pydantic import BaseModel


class PollOptionForm(BaseModel):
    text: str


class PollForm(BaseModel):
    title: str
    description: str
    options: List[PollOptionForm]


class PollStatistics(BaseModel):
    distribution: Dict[str, int]


class PollOptionResponse(BaseModel):
    id: str
    text: str
    selected: bool


class PollResponse(BaseModel):
    id: str
    title: str
    description: Optional[str]
    completed: bool
    can_edit: bool
    options: List[PollOptionResponse]
    statistics: PollStatistics


class PollCompletionRequest(BaseModel):
    option_id: str
